src/NSwag.SwaggerGeneration.AspNetCore/OperationParameterProcessor.cs

```diff
@@ -0,0 +1,334 @@
+                    if (attributes.All(a => a.GetType().Name != "SwaggerIgnoreAttribute" && a.GetType().Name != "JsonIgnoreAttribute"))
+                    {
+                        var fromQueryAttribute = attributes.SingleOrDefault(a => a.GetType().Name == "FromQueryAttribute");
+                        var propertyName = fromQueryAttribute.TryGetPropertyValue<string>("Name") ??
+                            context.SchemaGenerator.GetPropertyName(null, property);
+
+                        dynamic fromRouteAttribute = attributes.SingleOrDefault(a => a.GetType().FullName == "Microsoft.AspNetCore.Mvc.FromRouteAttribute");
+                        if (fromRouteAttribute != null && !string.IsNullOrEmpty(fromRouteAttribute?.Name))
+                            propertyName = fromRouteAttribute?.Name;
+
+                        dynamic fromHeaderAttribute = attributes.SingleOrDefault(a => a.GetType().FullName == "Microsoft.AspNetCore.Mvc.FromHeaderAttribute");
+                        if (fromHeaderAttribute != null && !string.IsNullOrEmpty(fromHeaderAttribute?.Name))
+                            propertyName = fromHeaderAttribute?.Name;
+
+                        var propertySummary = await property.GetXmlSummaryAsync().ConfigureAwait(false);
+                        var operationParameter = await context.SwaggerGenerator.CreatePrimitiveParameterAsync(
+                            propertyName, propertySummary, property.PropertyType, attributes).ConfigureAwait(false);
+
+                        // TODO: Check if required can be controlled with mechanisms other than RequiredAttribute
+
+                        var parameterInfo = _settings.ReflectionService.GetDescription(property.PropertyType, attributes, _settings);
+                        var isFileArray = IsFileArray(property.PropertyType, parameterInfo);
+                        if (parameterInfo.Type == JsonObjectType.File || isFileArray)
+                            InitializeFileParameter(operationParameter, isFileArray);
+                        else if (fromRouteAttribute != null
+                            || httpPath.ToLowerInvariant().Contains("{" + propertyName.ToLower() + "}")
+                            || httpPath.ToLowerInvariant().Contains("{" + propertyName.ToLower() + ":"))
+                        {
+                            operationParameter.Kind = SwaggerParameterKind.Path;
+                            operationParameter.IsNullableRaw = false;
+                            operationParameter.IsRequired = true; // Path is always required => property not needed
+                        }
+                        else if (fromHeaderAttribute != null)
+                            operationParameter.Kind = SwaggerParameterKind.Header;
+                        else
+                            operationParameter.Kind = SwaggerParameterKind.Query;
+
+                        operation.Parameters.Add(operationParameter);
+                    }
```

Comment:
@pranavkm  - this really seems like stuff that ApiDescription/ActionDescriptor should include already